### PR TITLE
chore: enable caching for all panel types in metrics v4

### PR DESCRIPTION
**Pull request by srikanthccv**
#### Screenshots

NA

### CODE REVIEW

File: pkg/query-service/app/queryBuilder/query_builder.go [hunk 1]
```diff
@@ -322,13 +322,12 @@ func isLogExpression(expression *govaluate.EvaluableExpression, params *v3.Query
 func (c *cacheKeyGenerator) GenerateKeys(params *v3.QueryRangeParamsV3) map[string]string {
 	keys := make(map[string]string)
 
-	// For non-graph panels, we don't support caching
-	if params.CompositeQuery.PanelType != v3.PanelTypeGraph {
-		return keys
-	}
-
```
Comment (srikanthccv) [inline pkg/query-service/app/queryBuilder/query_builder.go:329]: This is the main change. We used to return empty keys for every non-graph panel type. We still do it, but only when

1. query type is promql
2. The data source is logs

Now when it comes to metrics, things change

1. v3 - same as logs, only graph queries can be cached
2. v4 - everything can be cached

Comment (srikanthccv) [reply pkg/query-service/app/queryBuilder/query_builder.go:329]: We can't enable caching for other panels for logs and traces due to how their queries work. Unlike metrics, log queries don't return time series data that can be easily aggregated in query service. Instead, they fetch a single aggregated value for the entire time window. This makes it impossible to cache results for future use, like p95, avg, etc. We can support caching for simpler operations like sum or count, but we'd need to implement that first before enabling it here.

Comment (srikanthccv) [reply pkg/query-service/app/queryBuilder/query_builder.go:329]: Also there's no v4 for logs or traces - they work the same way.

Comment (ankitnayan) [reply pkg/query-service/app/queryBuilder/query_builder.go:329]: ok

File: pkg/query-service/app/queryBuilder/query_builder.go [hunk 2]
```diff
@@ -383,6 +387,10 @@ func (c *cacheKeyGenerator) GenerateKeys(params *v3.QueryRangeParamsV3) map[stri
 		} else if query.Expression == queryName && query.DataSource == v3.DataSourceMetrics {
 			var parts []string
 
+			if params.Version != "v4" && params.CompositeQuery.PanelType != v3.PanelTypeGraph {
```
Comment (nityanandagohain) [inline pkg/query-service/app/queryBuilder/query_builder.go:390]: please add a code comment here stating that, for metring if the version is v3 then we only support caching for Graph.

Will be easier to look back

Comment (srikanthccv) [reply pkg/query-service/app/queryBuilder/query_builder.go:402]: Added in https://github.com/SigNoz/signoz/pull/5651/commits/6d845923899135c4f7df6a161164deaf2ddb717b

**Review by nityanandagohain — APPROVED** [review]

**Pull request merged**